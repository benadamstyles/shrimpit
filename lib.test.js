const Shrimpit = require('./lib')

const inputPaths = (...arr) => ['', '', ...arr, '--json']
const outputSnapshot = async shrimpit => JSON.parse(await shrimpit.exec())

it('should create a snapshot of the files tree and of the unused exports', async () => {
  const shrimpit = new Shrimpit(inputPaths('test'))
  expect(await outputSnapshot(shrimpit)).toMatchSnapshot()
})

describe('globs', () => {
  it('should handle globs correctly', async () => {
    const shrimpit = new Shrimpit(inputPaths('test/**/*.js', '!**/(c|d)/*'))
    expect(await outputSnapshot(shrimpit)).toMatchSnapshot()
  })
})

describe('import * from ...', () => {
  it('should mark used properties as used exports', async () => {
    const shrimpit = new Shrimpit(inputPaths('test/import-all/*.js'))
    expect((await outputSnapshot(shrimpit))['Unused exports']).toMatchSnapshot()
  })
})
