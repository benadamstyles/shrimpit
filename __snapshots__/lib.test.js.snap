// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globs should handle globs correctly 1`] = `
Object {
  "Files tree": Object {
    "test": Object {
      "a": Object {
        "a.js": Object {
          "exports": Array [
            Object {
              "location": "test/a/a.js",
              "name": "a",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/a/a.js",
              "name": "c",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/a/a.js",
              "name": "d",
              "unnamedDefault": false,
            },
          ],
          "imports": Array [
            Object {
              "location": "test/b/b.js",
              "name": "test",
              "unnamedDefault": true,
            },
            Object {
              "location": "test/b/b.js",
              "name": "a",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/c/c.js",
              "name": "User",
              "unnamedDefault": false,
            },
          ],
        },
      },
      "b": Object {
        "b.js": Object {
          "exports": Array [
            Object {
              "location": "test/b/b.js",
              "name": "a",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/b/b.js",
              "name": "b",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/b/b.js",
              "name": "default (unnamed)",
              "unnamedDefault": true,
            },
          ],
          "imports": Array [
            Object {
              "location": "test/c/c.js",
              "name": "Cat",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/a/a.js",
              "name": "test",
              "unnamedDefault": true,
            },
            Object {
              "location": "test/d/d.js",
              "name": "unamedFunction",
              "unnamedDefault": true,
            },
          ],
        },
      },
      "import-all": Object {
        "exports.js": Object {
          "exports": Array [
            Object {
              "location": "test/import-all/exports.js",
              "name": "first",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/import-all/exports.js",
              "name": "second",
              "unnamedDefault": false,
            },
          ],
          "imports": Array [],
        },
        "imports.js": Object {
          "exports": Array [],
          "imports": Array [
            Object {
              "location": "test/import-all/exports.js",
              "name": "numbers",
              "unnamedDefault": true,
            },
          ],
        },
      },
    },
  },
  "Unused exports": Array [
    Object {
      "location": "test/a/a.js",
      "name": "a",
      "unnamedDefault": false,
    },
    Object {
      "location": "test/a/a.js",
      "name": "c",
      "unnamedDefault": false,
    },
    Object {
      "location": "test/a/a.js",
      "name": "d",
      "unnamedDefault": false,
    },
    Object {
      "location": "test/b/b.js",
      "name": "b",
      "unnamedDefault": false,
    },
  ],
}
`;

exports[`should create a snapshot of the files tree and of the unused exports 1`] = `
Object {
  "Files tree": Object {
    "test": Object {
      "a": Object {
        "a.js": Object {
          "exports": Array [
            Object {
              "location": "test/a/a.js",
              "name": "a",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/a/a.js",
              "name": "c",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/a/a.js",
              "name": "d",
              "unnamedDefault": false,
            },
          ],
          "imports": Array [
            Object {
              "location": "test/b/b.js",
              "name": "test",
              "unnamedDefault": true,
            },
            Object {
              "location": "test/b/b.js",
              "name": "a",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/c/c.js",
              "name": "User",
              "unnamedDefault": false,
            },
          ],
        },
      },
      "b": Object {
        "b.js": Object {
          "exports": Array [
            Object {
              "location": "test/b/b.js",
              "name": "a",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/b/b.js",
              "name": "b",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/b/b.js",
              "name": "default (unnamed)",
              "unnamedDefault": true,
            },
          ],
          "imports": Array [
            Object {
              "location": "test/c/c.js",
              "name": "Cat",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/a/a.js",
              "name": "test",
              "unnamedDefault": true,
            },
            Object {
              "location": "test/d/d.js",
              "name": "unamedFunction",
              "unnamedDefault": true,
            },
          ],
        },
      },
      "c": Object {
        "c.js": Object {
          "exports": Array [
            Object {
              "location": "test/c/c.js",
              "name": "Cat",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/c/c.js",
              "name": "User",
              "unnamedDefault": false,
            },
          ],
          "imports": Array [
            Object {
              "location": "test/a/a.js",
              "name": "a",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/a/a.js",
              "name": "c",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/a/a.js",
              "name": "d",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/b/b.js",
              "name": "b",
              "unnamedDefault": false,
            },
          ],
        },
      },
      "d": Object {
        "d.js": Object {
          "exports": Array [
            Object {
              "location": "test/d/d.js",
              "name": "test/d",
              "unnamedDefault": true,
            },
          ],
          "imports": Array [],
        },
      },
      "import-all": Object {
        "exports.js": Object {
          "exports": Array [
            Object {
              "location": "test/import-all/exports.js",
              "name": "first",
              "unnamedDefault": false,
            },
            Object {
              "location": "test/import-all/exports.js",
              "name": "second",
              "unnamedDefault": false,
            },
          ],
          "imports": Array [],
        },
        "imports.js": Object {
          "exports": Array [],
          "imports": Array [
            Object {
              "location": "test/import-all/exports.js",
              "name": "numbers",
              "unnamedDefault": true,
            },
          ],
        },
      },
    },
  },
  "Unused exports": Array [],
}
`;
